doc/source/dev/contributor/rendering_documentation.rst

```diff
@@ -217,6 +217,70 @@ Some examples:
         Some text that follows the list.
 
 
+Self-contained examples
+~~~~~~~~~~~~~~~~~~~~~~~
+All examples must be self-contained. It means all imports must be explicit,
+the data used must be defined in the examples, etc.
+
+    Yes::
+
+        >>> import numpy as np
+        >>> rng = np.random.default_rng()
+
+    No::
+
+        >>> rng = np.random.default_rng()
+
+The scope is defined
```

Comment:
I don't understand this part about scope. I do understand the part about interspersing code and explanations. If they refer to the same idea, maybe the part about scope can be removed. If not, "i.e." is probably not appropriate.

Reply:
I think that can be removed from here and clarified above, where each example is said to be self-contained. I'll make a suggestion.